### PR TITLE
Add SYSTIMER ETM

### PR DESCRIPTION
**Pull request by bjoernQ**
Adds ETM functionality to SYSTIMER

### CODE REVIEW

File: esp32c6-hal/examples/etm_blinky_systimer.rs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    gpio::{etm::GpioEtmChannels, IO},
+    peripherals::Peripherals,
+    prelude::*,
+    systimer::{etm::SysTimerEtmEvent, SystemTimer},
+};
+use esp_backtrace as _;
+
+#[entry]
+fn main() -> ! {
+    esp_println::logger::init_logger_from_env();
```
Comment (bugadani) [inline esp32c6-hal/examples/etm_blinky_systimer.rs:18]: Please put these behind a `#[cfg(feature = "log")]`. I just cleaned up like 26 of these in https://github.com/esp-rs/esp-hal/pull/810 and that isn't a fun activity

Comment (bugadani) [reply esp32c6-hal/examples/etm_blinky_systimer.rs:18]: Also please add the examples to CI

Comment (bjoernQ) [reply esp32c6-hal/examples/etm_blinky_systimer.rs:18]: ah I keep forgetting about the log vs defmt thing 👍 most probably will just remove it

Comment (bugadani) [reply esp32c6-hal/examples/etm_blinky_systimer.rs:18]: Normally this should fail in CI but these examples aren't checked (yet) ;)

Comment (bjoernQ) [reply esp32c6-hal/examples/etm_blinky_systimer.rs:18]: we do `cargo build --examples`? there are no features needed so it should build it

Comment (bugadani) [reply esp32c6-hal/examples/etm_blinky_systimer.rs:18]: True enough. esp-println's `log` is currently enabled by default right now which makes this pass...

**Review by JurajSadel — APPROVED** [review]
LGTM!

**Pull request merged**